### ISSUE TITLE
Services should use keycloak auth

### ISSUE THREAD

**Issue by rocketeerbkw** [uselagoon/lagoon#1017]
In order to move all auth/authz to keycloak, lagoon services should start authenticating to keycloak to get JWTs that are compatible with the API instead of generating the custom "admin" JWTs.

- Create ServiceAccount for each Project and add JWT of this ServiceAccount into running containers so they can use that JWT to authenticate at Lagon API (example: drush)
- Convert lagoon services to use keycloak service accounts instead of custom JWT

**Comment by bomoko** [comment]
Still a good idea, and should be implemented.

**Comment by tobybellwood** [comment]
IMO keycloak auth is the correct place to authenticate user activity. Internal service activity should probably be authenticated internally to Lagoon. Replacing the JWTs with an internal REST API using a shared secret token would likely be faster and less reliant on Keycloak.

**Comment by rocketeerbkw** [comment]
What about authorization? Currently services using the legacy token and services we've already implemented with keycloak have "admin" rights. It would be best practice to assign the minimum required rights to each service that needs it. If we have a totally separate system for authentication we'd still need to query keycloak for authorization.

**Comment by smlx** [comment]
I agree with @rocketeerbkw that we should be using Keycloak for this. In fact [OAuth 2.0 has a flow](https://www.rfc-editor.org/rfc/rfc6749#section-4.4) specifically for this scenario. The [Auth0 documentation](https://auth0.com/docs/get-started/authentication-and-authorization-flow/client-credentials-flow) has a great explanation and diagram.
